File: TwoPointers/csharp/IsPalindrome.cs
namespace TwoPointersSolutions;

public static class IsPalindrome
{
    public static bool Check(string s)
    {
        var i = 0;
        var j = s.Length - 1;
        while (i < j)
        {
            if (s[i] != s[j])
            {
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
}

File: TwoPointers/java/isPalindrome.java
public final class IsPalindrome {
  private IsPalindrome() {
  }

  public static boolean check(String s) {
    int i = 0;
    int j = s.length() - 1;
    while (i < j) {
      if (s.charAt(i) != s.charAt(j)) {
        return false;
      }
      i++;
      j--;
    }
    return true;
  }
}
